### DISCUSSION TITLE
Is there a geopandas alternative for image stacks ?

### DISCUSSION THREAD

**Discussion by 12rambau** [geopandas/geopandas#2888] [General]
I entered the geospatial analysis through Google Earth Engine and moved to open-source afterward as I was an enthousiast python dev. 
I'm not validating everything they did but GEE clearly set up new standards for Geo analysis. 

Geopandas is the equivalent of ee.FeatureCollection but is there an equivalent to ee.ImageCollection ? What I mean is: a stack of image (I assume stored in a vrt) with metadata stored in pandas columns. This could allow to performed pixel based computation and filtering super easily. 

If it's not existing is it something you think worth working on ?

**Comment by martinfleis** [comment]
The raster side of things is usually dealt with using `xarray` and other packages within its ecosystem like `rioxarray`. The ecosystem around it is quite large and robust so check that and a related [pangeo](https://pangeo.io) project for more info. 

While you can use `rasterio` and pure numpy arrays, I would strongly recommend going the xarray path as it is developed (among other) for this purpose.

**Reply by 12rambau** [reply]
Thank you very much I had already used rioxarray in the past but I didn't know it was possible to agregate multiple TIff (a mosaic for example.

**Comment by theroggy** [comment]
Depending on your exact needs, you could also have a look at https://openeo.org. This is being developed as an open (source) alternative to google earth engine, including the aspect of running the processing (typically) on the server, close to the data.

It is not as mature yet as gee, and I haven't used it a lot (yet), but e.g. to create periodic mosaics of sentinel images like I have done in the past using gee it worked really similar.
